searx/engines/niconico.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Niconico search engine for searxng"""
+
+import time
+
+from urllib.parse import urlencode
+from datetime import datetime, timedelta
+from lxml import html
+
+from searx.utils import extract_text
+
+about = {
+    "website": "https://www.nicovideo.jp/",
+    "wikidata_id": "Q697233",
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": "HTML",
+    "language": "ja",
+}
+
+categories = ["videos"]
+paging = True
+
+time_range_support = True
+time_range_dict = {"day": 86400, "week": 604800, "month": 2592000, "year": 31536000}
+
+base_url = "https://www.nicovideo.jp"
+
+
+def request(query, params):
+    query_params = {"page": params['pageno']}
+
+    if time_range_dict.get(params['time_range']):
+        now = int(time.time())
+        past = now - time_range_dict[params["time_range"]]
+
+        query_params['end'] = time.strftime('%Y-%m-%d', time.gmtime(now))
+        query_params['start'] = time.strftime('%Y-%m-%d', time.gmtime(past))
+
+    params['url'] = f"{base_url}/search/{query}?{urlencode(query_params)}"
+    return params
+
+
+def response(resp):
+    results = []
+    dom = html.fromstring(resp.text)
+
+    for item in dom.xpath('//li[@data-video-item]'):
+        relative_url = extract_text(item.xpath('.//a[@class="itemThumbWrap"]/@href')[0])
```

Comment:
Please use `eval_xpath` from `searx.utils` instead of `item.xpath` everywhere here.